### PR TITLE
treat 0 bucket bound as an error.

### PR DESCRIPTION
**Pull request by rghetia**
Enforcing [spec](https://github.com/census-instrumentation/opencensus-proto/blob/c9cc5de7251f8a320dd6cb4f92e5698ba86f284c/src/opencensus/proto/metrics/v1/metrics.proto#L222-L225)

### CODE REVIEW

File: stats/view/view.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func (v *View) canonicalize() error {
 	})
 	sort.Float64s(v.Aggregation.Buckets)
 	for _, b := range v.Aggregation.Buckets {
-		if b < 0 {
+		if b <= 0 {
 			return ErrNegativeBucketBounds
```
Comment (songy23) [inline stats/view/view.go]: This should be renamed "ErrNonPositiveBucketBounds" now.

Comment (rghetia) [reply stats/view/view.go]: done.

Comment (rghetia) [reply stats/view/view.go]: I changed it back as I am silently dropping 0 bound bucket.

Comment (songy23) [reply stats/view/view.go]: OK, silently dropping it makes more sense.

File: stats/view/view_test.go [hunk 2]
```diff
@@ -459,18 +459,6 @@ func TestViewRegister_negativeBucketBounds(t *testing.T) {
 	}
 }
 
-func TestViewRegister_zeroBucketBounds(t *testing.T) {
```
Comment (songy23) [inline stats/view/view_test.go]: Consider adding a test on verifying the leading 0 bucket bounds are dropped.

Comment (rghetia) [reply stats/view/view_test.go]: added the test.

**Comment by odeke-em** [comment]
Oh oh this is a little concerning and going to be a breaking change for many people and all our libraries, examples and integrations too. For more than a year we've had buckets of `[0, 10` etc acceptable. Enforcing a spec is great but should be gradual IMHO.

**Comment by rghetia** [comment]
@odeke-em it will still accept 0 value bucket but will silently drop. So it will not break the api.

**Pull request merged**